### PR TITLE
refactor: move to modules, ox_lib, style fixes

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

- Moves to modules
- Further implementation of ox_lib
- Style fixes

Special thanks to mockdot for his help with the ignore.lua addeventhandler.

I did not include a fix for the carwash as this is already done in [#33](https://github.com/Qbox-project/qbx-smallresources/pull/33). My apologies again.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

File: client/fireworks.lua [hunk 1]
```diff
@@ -71,60 +54,51 @@ local function DoFireWork(asset, coords)
...
+        'proj_xmas_firework',
+        'proj_indep_firework_v2',
+        'proj_indep_firework'
+    }
+    
+    for i = 1, #assets do
+        local asset = assets[i]
+        if not HasNamedPtfxAssetLoaded(asset) then
+            lib.requestNamedPtfxAsset(asset)
         end
```
Comment (Manason) [inline client/fireworks.lua]:
```suggestion
        lib.requestNamedPtfxAsset(assets[i])
```

File: client/consumables.lua [hunk 2]
```diff
@@ -151,253 +149,319 @@ end
...
+            car = false,
+            mouse = false,
+            combat = true
+        }
+    }) then -- if completed
         TriggerEvent('inventory:client:ItemBox', QBCore.Shared.Items[itemName], 'remove')
         exports.scully_emotemenu:cancelEmote()
-        TriggerServerEvent('consumables:server:addThirst', QBCore.Functions.GetPlayerData().metadata['thirst'] + ConsumablesDrink[itemName])
-    end)
+        TriggerServerEvent('consumables:server:addThirst', PlayerData.metadata.thirst + Config.Consumables.drink[itemName])
```
Comment (Manason) [inline client/consumables.lua]: Does Config.Consumables.drink exist?

File: client/consumables.lua [hunk 3]
```diff
@@ -151,253 +149,319 @@ end
...
+            car = false,
+            mouse = false,
+            combat = true
+        }
+    }) then -- if completed
         exports.scully_emotemenu:cancelEmote()
         TriggerEvent('inventory:client:ItemBox', QBCore.Shared.Items[itemName], 'remove')
         TriggerServerEvent('consumables:server:drinkAlcohol', itemName)
-        TriggerServerEvent('consumables:server:addThirst', QBCore.Functions.GetPlayerData().metadata['thirst'] + ConsumablesAlcohol[itemName])
+        TriggerServerEvent('consumables:server:addThirst', PlayerData.metadata.thirst + Config.Consumables.alcohol[itemName])
```
Comment (Manason) [inline client/consumables.lua]: ditto

File: client/consumables.lua [hunk 4]
```diff
@@ -151,253 +149,319 @@ end
...
         elseif alcoholCount >= 4 then
             TriggerEvent('evidence:client:SetStatus', 'heavyalcohol', 200)
         end
-
-    end, function() -- Cancel
+    else -- if canceled
         exports.scully_emotemenu:cancelEmote()
-        QBCore.Functions.Notify('Cancelled..', 'error')
-    end)
+        QBCore.Functions.Notify('Canceled...,', 'error')
```
Comment (Manason) [inline client/consumables.lua]:
```suggestion
        QBCore.Functions.Notify('Canceled...', 'error')
```

File: client/consumables.lua [hunk 5]
```diff
@@ -151,253 +149,319 @@ end
...
+            combat = true
+        },
+        anim = {
+            dict = 'mp_suicide',
+            clip = 'pill',
+            flag = 49
+        }
+    }) then -- if completed
         TriggerServerEvent('consumables:server:useXTCBaggy')
         TriggerEvent('inventory:client:ItemBox', QBCore.Shared.Items['xtcbaggy'], 'remove')
```
Comment (Manason) [inline client/consumables.lua]:
```suggestion
        TriggerEvent('inventory:client:ItemBox', QBCore.Shared.Items.xtcbaggy, 'remove')
```

File: client/vehicletext.lua [hunk 6]
```diff
@@ -1,15 +1,13 @@
...
+        end
+        if v.hash ~= 0 and v.hash ~= nil then
+			AddTextEntryByHash(v.hash, text)
 		end
-		if v['hash'] ~= 0 and v['hash'] ~= nil then
-			AddTextEntryByHash(v['hash'],text)
-		end
-	end
-end)
+    end
```
Comment (Manason) [inline client/vehicletext.lua]: fix indenting

File: fxmanifest.lua [hunk 7]
```diff
@@ -1,11 +1,16 @@
 fx_version 'cerulean'
 game 'gta5'
 
-description 'QB-SmallResources'
+description 'QBX-SmallResources'
```
Comment (Manason) [inline fxmanifest.lua]:
```suggestion
description 'Collection of small scripts'
```

File: fxmanifest.lua [hunk 8]
```diff
@@ -18,5 +23,11 @@ files {
 	'relationships.dat'
 }
 
+modules {
+	'qbx-core:core',
+    'qbx-core:playerdata',
+    'qbx-core:utils'
+}
```
Comment (Manason) [inline fxmanifest.lua]: fix indentation?

**Review by Manason — CHANGES_REQUESTED** [review]

**Comment by mafewtm** [comment]
Merged into main to correct a conflict [#35](https://github.com/Qbox-project/qbx-smallresources/pull/35) produced with this PR when fixing ignore.lua

File: client/ignore.lua [hunk 9]
```diff
@@ -11,9 +11,9 @@ CreateThread(function()
 end)
 
 AddEventHandler('populationPedCreating', function(x, y, z)
-    Wait(500)    -- Give the entity some time to be created
-    local _, handle = GetClosestPed(vec3(x, y, z), {}) -- Get the entity handle
-    SetPedDropsWeaponsWhenDead(handle, false)
+	Wait(500)	-- Give the entity some time to be created
+	local handle = GetClosestPed(vec3(x, y, z)) -- Get the entity handle
+	SetPedDropsWeaponsWhenDead(handle, false)
```
Comment (Manason) [inline client/ignore.lua]: This looks indented too much?

File: fxmanifest.lua [hunk 10]
```diff
@@ -25,9 +25,9 @@ files {
 
 modules {
 	'qbx-core:core',
-    'qbx-core:playerdata',
-    'qbx-core:utils'
+	'qbx-core:playerdata',
+	'qbx-core:utils'
```
Comment (Manason) [inline fxmanifest.lua]: These look indented too much

**Review by Manason — CHANGES_REQUESTED** [review]

File: client/ignore.lua (hunk 9 above)
Comment (mafewtm) [reply client/ignore.lua]: This one was done in the pr you just merged but I believe these are all 4 space indentations

**Comment by Manason** [comment]
Fix conflicts

**Comment by mafewtm** [comment]
Merged into main to correct a conflict [#36](https://github.com/Qbox-project/qbx-smallresources/pull/36) produced with this PR when updating commands to OX which were already done in this PR.

**Pull request merged**